src/components/DynamicList/FormAddFood.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from "react";
+
+class FormAddFood extends Component {
+  state = {
+    name: "",
+    calories: 0,
+    image: "/images/default.png",
+    quantity: 0
+  };
+
+  handlerFormSubmit = event => {
+    event.preventDefault();
+    this.props.handleAddFood(this.state);
+    /* and reset */
+    this.setState({ name: "", calories: 0, image: "", quantity: 0 });
+    this.props.OnOffvisibleSearchNFindFood();
+    this.props.OnOffAddfood();
+  };
+
+  handleOnOffFormShow = () => {
+    const { visible } = this.state;
+    this.setState({ visible: !visible });
+  };
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  render() {
+    const { name, calories, image } = this.state;
+    const { OnOffvisibleSearchNFindFood, OnOffAddfood } = this.props;
+
+    return (
+      <div>
+        <form onSubmit={this.handlerFormSubmit}>
+          <label>Name:</label>
+          <input
+            type="text"
+            name="name"
+            value={name}
+            onChange={e => this.handleChange(e)}
```

Comment:
No necesitas hacer `onChange={e => this.handleChange(e)}` para pasarle el evento a la función, ya lo hace intrinsecamente: `onChange={this.handleChange}`

Reply:
cierto